src/interface_test.jl

```diff
@@ -0,0 +1,34 @@
+# This file is a part of DensityInterface.jl, licensed under the MIT License (MIT).
+
+"""
+    DensityInterface.test_density_interface(d, x, ref_logd_at_x)
+
+Test if `d` is compatible with `DensityInterface`.
+
+Tests that [`logdensityof(d, x)`](@ref) equals `ref_logd_at_x` and
+that the behavior of [`logdensityof(d)`](@ref),
+[`densityof(d, x)`](@ref) and [`densityof(d)`](@ref) is consistent.
+
+Also tests if `logfuncdensity(logdensityof(d))` returns
+a density equivalent to `d` in respect to the functions above.
```

Comment:
Maybe also check that `logfuncdensity` and `logdensityof` are defined as imverses correctly (since one could spevialize `logfuncdensity` as well)?

Reply:
You mean using `test_inverse(logdensityof, d; compare=...; ...)`? I had thought about it, but `logdensityof` and `logfuncdensity` operate on objects/functions, not values, so I think that could fail too easily. Users won't always define equality properly on such objects. So I test for equivalent behavior instead.